staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/types_jsonschema.go

```diff
@@ -18,13 +18,16 @@ package v1beta1
 
 // JSONSchemaProps is a JSON-Schema following Specification Draft 4 (http://json-schema.org/).
 type JSONSchemaProps struct {
-	ID                   string                     `json:"id,omitempty" protobuf:"bytes,1,opt,name=id"`
-	Schema               JSONSchemaURL              `json:"$schema,omitempty" protobuf:"bytes,2,opt,name=schema"`
-	Ref                  *string                    `json:"$ref,omitempty" protobuf:"bytes,3,opt,name=ref"`
-	Description          string                     `json:"description,omitempty" protobuf:"bytes,4,opt,name=description"`
-	Type                 string                     `json:"type,omitempty" protobuf:"bytes,5,opt,name=type"`
-	Format               string                     `json:"format,omitempty" protobuf:"bytes,6,opt,name=format"`
-	Title                string                     `json:"title,omitempty" protobuf:"bytes,7,opt,name=title"`
+	ID          string        `json:"id,omitempty" protobuf:"bytes,1,opt,name=id"`
+	Schema      JSONSchemaURL `json:"$schema,omitempty" protobuf:"bytes,2,opt,name=schema"`
+	Ref         *string       `json:"$ref,omitempty" protobuf:"bytes,3,opt,name=ref"`
+	Description string        `json:"description,omitempty" protobuf:"bytes,4,opt,name=description"`
+	Type        string        `json:"type,omitempty" protobuf:"bytes,5,opt,name=type"`
+	Format      string        `json:"format,omitempty" protobuf:"bytes,6,opt,name=format"`
+	Title       string        `json:"title,omitempty" protobuf:"bytes,7,opt,name=title"`
+	// default is a default value for undefined object fields.
+	// Defaulting is an alpha feature under the CustomResourceDefaulting feature gate.
+	// Defaulting required spec.preserveUnknownFields to be false.
```

Comment:
nit: s/required/requires/
add `+optional`?

Reply:
all of these are optional. I would rather prefer a follow-up PR adding docs for all the fields here. All this is totally undocumented.